File: Code/C_Code/03.1.1_LightWater/LightWater.c
/**********************************************************************
* Filename    : LightWater.c
* Description : Display 10 LEDBar Graph 
* Author      : freenove
* modification: 2016/06/13
**********************************************************************/
#include <wiringPi.h>
#include <stdio.h>
#define leds 10
int pins[leds] = {0,1,2,3,4,5,6,8,9,10};
void led_on(int n)//make led_n on
{
	digitalWrite(n, LOW);
}

void led_off(int n)//make led_n off
{
	digitalWrite(n, HIGH);
}

int main(void)
{
	int i;
	printf("Program is starting ... \n");
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}
	for(i=0;i<leds;i++){       //make leds pins' mode is output
		pinMode(pins[i], OUTPUT);
	}
	while(1){
		for(i=0;i<leds;i++){   //make led on from left to right
			led_on(pins[i]);
			delay(100);
			led_off(pins[i]);
		}
		for(i=leds-1;i>-1;i--){   //make led on from right to left
			led_on(pins[i]);
			delay(100);
			led_off(pins[i]);
		}
	}
	return 0;
}

File: Code/Python_Code/03.1.1_LightWater/LightWater.py
#!/usr/bin/env python3
########################################################################
# Filename    : LightWater.py
# Description : Use LEDBar Graph(10 LED) 
# auther      : www.freenove.com
# modification: 2019/12/28
########################################################################
import RPi.GPIO as GPIO
import time

ledPins = [11, 12, 13, 15, 16, 18, 22, 3, 5, 24]

def setup():    
    GPIO.setmode(GPIO.BOARD)        # use PHYSICAL GPIO Numbering
    GPIO.setup(ledPins, GPIO.OUT)   # set all ledPins to OUTPUT mode
    GPIO.output(ledPins, GPIO.HIGH) # make all ledPins output HIGH level, turn off all led

def loop():
    while True:
        for pin in ledPins:     # make led(on) move from left to right
            GPIO.output(pin, GPIO.LOW)  
            time.sleep(0.1)
            GPIO.output(pin, GPIO.HIGH)
        for pin in ledPins[::-1]:       # make led(on) move from right to left
            GPIO.output(pin, GPIO.LOW)  
            time.sleep(0.1)
            GPIO.output(pin, GPIO.HIGH)

def destroy():
    GPIO.cleanup()                     # Release all GPIO

if __name__ == '__main__':     # Program entrance
    print ('Program is starting...')
    setup()
    try:
        loop()
    except KeyboardInterrupt:  # Press ctrl-c to end the program.
        destroy()

